pkg/cri/server/image_pull.go

```diff
@@ -136,7 +137,38 @@ func (c *criService) PullImage(ctx context.Context, r *runtime.PullImageRequest)
 			containerd.WithChildLabelMap(containerdimages.ChildGCLabelsFilterLayers))
 	}
 
-	image, err := c.client.Pull(ctx, ref, pullOpts...)
+	// Check if there's currently an ongoing pull of the same image, and if so just wait on the result.
+	var (
+		bc     *imageBroadcaster
+		loaded bool
+	)
+	if c.config.DisableSameImageParallelPull {
+		bc, loaded = c.imagesInProgress.loadOrStore(ref)
```

Comment:
I think we shouldn't rely on the ref for identifying images because images having same ref doesn't guarantee that their contents are the same.

Reply:
Yea that's a good point. I thought as this was mainly to counter act pulling the same image in parallel, that for an image to get updated in that timeframe would seem unlikely, but not impossible. 

e.g. Somewhere in the span of pulling the same ref/image 3 times I suppose this could happen.
Pull myimage:latest
Pull myimage:latest
Image updated on registry
Pull myimage:latest <- Without this change we'd start fetching the new content, but now we don't as long as there's any in progress at the same time.

Reply:
@ktock You think "ref@index/manifest" is good enough?